README.rst

```diff
@@ -36,6 +36,21 @@ You can then call pip with::
 and this will then install the requested packages and all dependencies,
 ignoring any releases after the cutoff date specified above.
 
+For convenience, it is also possible to specify the cutoff date as part of the
+repository URL, using the pattern http://{host}/cutoff/{cutoff_date}/.
+For example::
+
+   pip install --index-url http://127.0.0.1:5000/snapshot/2024-12-03 astropy
```

Comment:
Should this command be pypi-timemachine?

Reply:
I think I need your help to make the docs clearer though :joy: 

Basically:

```
pypi-timemachine {CUTOFF_DATE}
```

Starts a server which, when you hit `http://localhost:5000/`, redirects to `http://localhost:5000/snapshot/{CUTOFF_DATE}`. In practice, we could choose to make the CLI argument entirely optional. There is no problem at all hitting a different snapshot date - it is fully dynamic

Reply:
Oh I see! Then I think we might want to indeed support starting pypi-timemachine without any arguments as would be clearer?

Reply:
OK, noted in https://github.com/astrofrog/pypi-timemachine/issues/25.

The main topic of this PR is to have a custom index URL (I was backfilling in the README items that I forgot to mention before). Suggest we do the default change separately.